core/src/main/scala/org/http4s/internal/CharPredicate.scala

```diff
@@ -149,7 +149,7 @@ object CharPredicate {
   private def unmaskable(c: Char) = c >= 128
 
   // efficient handling of 7bit-ASCII chars
-  case class MaskBased private[CharPredicate] (lowMask: Long, highMask: Long)
+  sealed case class MaskBased private[CharPredicate] (lowMask: Long, highMask: Long)
```

Comment:
I'm still not convinced about having `sealed` case classes. They shouldn't be extendable at all, I believe.

Reply:
Right. We can make this final since the private constructor already makes it effectively final.